### PR TITLE
[ENG-468] Contact Client fixes

### PR DESCRIPTION
**Pull request by cykonetic**

### CODE REVIEW

File: Event/ContactClientTimelineEvent.php [hunk 1]
```diff
@@ -184,6 +184,8 @@ public function __construct(
      */
     public function addEvent(array $data)
     {
+        $stop = 'here';
```
Comment (heathdutton) [inline Event/ContactClientTimelineEvent.php]: Please remove.

File: EventListener/ContactClientSubscriber.php [hunk 2]
```diff
@@ -166,6 +166,8 @@ public function onContactClientDelete(ContactClientEvent $event)
      */
     public function onTimelineGenerate(ContactClientTimelineEvent $event)
     {
+        $stop = 'here';
```
Comment (heathdutton) [inline EventListener/ContactClientSubscriber.php]: Please remove.

File: Model/ContactClientModel.php [hunk 3]
```diff
@@ -306,22 +306,28 @@ public function getStats(
         $dateFormat = null,
         $canViewOthers = true
     ) {
-        $unit           = (null === $unit) ? $this->getTimeUnitFromDateRange($dateFrom, $dateTo) : $unit;
-        $dateToAdjusted = clone $dateTo;
-        if (in_array($unit, ['H', 'i', 's'])) {
-            // draw the chart with the correct intervals for intra-day
-            $dateToAdjusted->setTime(23, 59, 59);
+        if ($dateFrom->getTimezone()->getName() !== 'UTC') {
+            $dateFrom->setTimezone(new \DateTimeZone('UTC'));
```
Comment (heathdutton) [inline Model/ContactClientModel.php]: Resetting to UTC invalidates the user's timezone. You should not need to do this.

File: Model/ContactClientModel.php [hunk 4]
```diff
@@ -306,22 +306,28 @@ public function getStats(
...
         }
-        $chart = new LineChart($unit, $dateFrom, $dateToAdjusted, $dateFormat);
-        $query = new ChartQuery($this->em->getConnection(), $dateFrom, $dateToAdjusted, $unit);
-        $stat  = new Stat();
+        $dateFrom->modify('midnight');
+
+        if ($dateTo->getTimezone()->getName() !== 'UTC') {
+            $dateTo->setTimezone(new \DateTimeZone('UTC'));
```
Comment (heathdutton) [inline Model/ContactClientModel.php]: Resetting to UTC invalidates the user's timezone. You should not need to do this.

File: Model/ContactClientModel.php [hunk 5]
```diff
@@ -306,22 +306,28 @@ public function getStats(
...
+        }
+        $dateTo->modify('tomorrow midnight -1 sec');
```
Comment (heathdutton) [inline Model/ContactClientModel.php]: You shouldn't have to do this... getTimeUnitFromDateRange adjusts the datetime provided by reference.
That's what `$dateToAdjusted = clone $dateTo;` was for.

File: Model/ContactClientModel.php (hunk 3 above)
Comment (cykonetic) [reply Model/ContactClientModel.php]: it was being done below this. I just updated the logic to do so?

**Pull request merged**